### PR TITLE
Update most shadows to support API 23

### PR DESCRIPTION
**Pull request by tmrudick**
Upgrading shadows and stubs to work with API 23.

Still need to implement RoboCookieManager without apache.http as well as verify the placement of AndroidHttpClient. 

This requires a new android-all jar to pick up changes in paths from L -> M. We also have to disable Jack when building to ensure that we have class files for leaf dependencies.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.robolectric.TestRunners;
-import org.apache.http.util.CharArrayBuffer;
+// import org.apache.http.util.CharArrayBuffer;
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java]: Remove commented out code.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java [hunk 2]
```diff
@@ -13,8 +13,8 @@
   public void append_shouldBeInstrumented() {
     // CharArrayBuffer.expand uses a modified version of java.lang.System.arraycopy.
     // The class needs to be instrumented and re-written in order to work correctly.
-    final CharArrayBuffer buffer = new CharArrayBuffer(1);
-    buffer.append("Foo");
-    assertThat(buffer.toString()).isEqualTo("Foo");
+    // final CharArrayBuffer buffer = new CharArrayBuffer(1);
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java]: Remove commented code

File: robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java [hunk 3]
```diff
@@ -147,10 +147,4 @@ public void unregisterCallback_shouldRemoveCallbacks() throws Exception {
     connectivityManager.unregisterNetworkCallback(callback1);
     assertEquals(0, shadowConnectivityManager.getNetworkCallbacks().size());
   }
-
-  @Test(expected=IllegalArgumentException.class) @Config(sdk = 21)
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java]: Why remove this test?

File: robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java (hunk 2 above)
Comment (tmrudick) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java]: Will do. Although, I think this is the only test testing System.arrayCopy is instrumented. Do you know of another place where arrayCopy is used that I could modify this test to use?

File: robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java (hunk 3 above)
Comment (tmrudick) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java]: The unregisterNetworkCallback method now has a version which takes a PendingIntent (http://developer.android.com/reference/android/net/ConnectivityManager.html#unregisterNetworkCallback(android.app.PendingIntent)). This makes passing in a single null argument ambiguous which prevents this from compiling under 23.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java (hunk 2 above)
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowCharArrayBufferTest.java]: Not sure sorry.

**Comment by erd** [comment]
> This requires a new android-all jar to pick up changes in paths from L -> M. We also have to disable Jack when building to ensure that we have class files for leaf dependencies.

I'll rebuild the android-all jar as soon as this is merged.

**Comment by tmrudick** [comment]
Removed the commented out code. A few things that still need to be done for 23 support:

* Find another way to test System.arrayCopy instrumentation
* Rewrite RoboCookieManager without using apache.http
* Fix remaining failing tests (CookieManager tests are failing, some XML inflation tests are also failing)
* New android-all jar (Thanks Erich!)

**Comment by erd** [comment]
I'll start working on a new android-all jar.

**Pull request merged**